volumes/vds/sql/create-function-aggregate_15min_vds_volumes.sql

```diff
@@ -0,0 +1,43 @@
+CREATE OR REPLACE FUNCTION vds.aggregate_15min_vds_volumes(_start_date timestamp, _end_date timestamp
+	)
+    RETURNS void
+    LANGUAGE 'plpgsql'
+
+    COST 100
+    VOLATILE SECURITY DEFINER 
+AS $BODY$
+
+BEGIN
+	
+    --Aggregated into speed bins and 1 hour bin
+    INSERT INTO vds.volumes_15min (divisionid, vdsid, detector_id, datetime_bin, volume_15min)
+    
+    SELECT 
+        d.divisionid,
+        d.vdsid,
+        c.detector_id,
+        d.datetime_15min,
+        SUM(d.volumevehiclesperhour) / 4 / 45 AS volume_15min
+    FROM vds.raw_vdsdata AS d
+    JOIN vds.vdsconfig AS c ON
+        d.vdsid = c.vdsid
+        AND d.divisionid = c.divisionid
+        AND d.datetime_15min >= c.starttimestamputc
+        AND (
+            d.datetime_15min <= c.endtimestamputc
+            OR c.endtimestamputc IS NULL) --no end date
+    WHERE 
+        datetime_15min >= _start_date --'2023-06-01 01:00:00'::timestamp 
+        AND datetime_15min < _end_date --'2023-06-01 02:00:00'::timestamp
+    GROUP BY
+        d.divisionid,
+        d.vdsid,
+        c.detector_id,
+        d.datetime_15min
```

Comment:
Did you already validate that each hour has 4 15-min hourly volume? Or should you add a condition with `HAVING COUNT(*) = 4` for instance?